### PR TITLE
Added discovery CLI tests

### PR DESCRIPTION
**Pull request by lzap**
Discovery 6.0 released showed our weak API/CLI test coverage and this is my
attempt to improve this. I will go ahead and add more tests if you like that.
CLI for 6.0 is currently broken (as well as for develop branch):

http://projects.theforeman.org/issues/15957

### CODE REVIEW

File: fb-setup-discovery.bats [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+set -o pipefail
+
+load os_helper
+load foreman_helper
+
+@test "install discovery plugin" {
+  yum -y install tfm-rubygem-foreman_discovery tfm-rubygem-hammer_cli_foreman_discovery
+}
+
+@test "create discover-host script" {
```
Comment (ekohl) [inline fb-setup-discovery.bats]: Can't this be written as a helper function?

Comment (lzap) [inline fb-setup-discovery.bats]: You want me to add `discovery_helper` shell script or create new function in the bats itself?

**Comment by lzap** [comment]
Added more CLI tests:

```
[root@cisco-b200m1-03 foreman-bats]# ./fb-setup-discovery.bats
 ✓ install discovery plugin
 ✓ restart foreman
 ✓ create discover-host script
 ✓ discover a host
 ✓ discover a second host
 ✓ list discovered hosts and refresh hammer cache
 ✓ find a discovered host id
 ✓ show discovered host
 ✓ provision discovered host
 ✓ create a rule
 ✓ list discovery rules
 ✓ show discovery rule
 ✓ find a discovered host id
 ✓ show discovered host
 ✓ auto provision a host

15 tests, 0 failures
```

File: fb-setup-discovery.bats (hunk 1 above)
Comment (lzap) [inline fb-setup-discovery.bats]: Ok that's done, ready for re-review.

Comment (ekohl) [inline fb-setup-discovery.bats]: You still create /usr/local/bin/discover-host. I don't really see why that's needed.

Comment (lzap) [inline fb-setup-discovery.bats]: > You still create /usr/local/bin/discover-host. I don't really see why that's needed.

Oh that's intentional, I very often sign in the tested box when it fails
and I test discovered hosts provisioning.

-- 
Later,
 Lukas #lzap Zapletal

Comment (ekohl) [inline fb-setup-discovery.bats]: To me that sounds like we provide insufficient debug info during the normal process. If we'd test this under Jenkins that sounds like something we'd want.

Comment (lzap) [inline fb-setup-discovery.bats]: > To me that sounds like we provide insufficient debug info during the normal process. If we'd test this under Jenkins that sounds like something we'd want.

Frankly, I use BATS to have "everyday" instances I play with. Having a
helper ready is quite handy.

-- 
Later,
 Lukas #lzap Zapletal

Comment (ekohl) [inline fb-setup-discovery.bats]: @domcleal / @mmoll what do you think?

Comment (domcleal) [inline fb-setup-discovery.bats]: I agree with @ekohl. Test scripts shouldn't be adding binaries, it should be a helper within the script.

Comment (lzap) [inline fb-setup-discovery.bats]: > I agree with @ekohl. Test scripts shouldn't be adding binaries, it should be a helper within the script.

Refactored that way.

-- 
Later,
 Lukas #lzap Zapletal

**Comment by mmoll** [comment]
merged, díky @lzap!

**Pull request merged**